### DISCUSSION TITLE
umi4 qiankun 微应用使用useMode通信，报dispatcher找不到的错误

### DISCUSSION THREAD

**Discussion by lbyupup** [umijs/umi#9191] [Q&A]
"@umijs/plugins": "^4.0.15",
"@umijs/max": "^4.0.0",
"@umijs/plugin-qiankun": "^2.24.1",
我在微应用中使用useModel，内部源码报错，求助下大佬
```ts
import { useModel } from 'umi';
const { data } = useModel('@@qiankunStateFromMaster') ?? {};
```
报错如下
![image](https://user-images.githubusercontent.com/72724939/187852448-633ea4cb-40f6-4873-a0eb-f5d27a8615bf.png)

**Comment by fz6m** [comment]
1. max 已经内置了 qiankun, model 等插件，无需安装，通过如下方式启用：

    ```ts
    // .umirc.ts
    
    export default {
      qiankun: {},
      model: {}
    }
    ```

2. 请删除多余的包：

    ```bash
      pnpm remove @umijs/plugins @umijs/plugin-qiankun
    ```

3. max 中，所有导入需要使用 `@umijs/max` 而不是 `umi`：

    ```diff
    -import some from 'umi'
    +import some from '@umijs/max'
    ```

**Reply by lbyupup** [reply]
删除掉这两个依赖 pnpm remove @umijs/plugins @umijs/plugin-qiankun ，使用下面代码导入
```ts
import { useModel } from '@umijs/max'
```
项目启动后仍热出现报错，config配置如下。是还需要什么地方配置一下吗
```ts
qiankun: {
    slave: {}
  },
  model: {},
```

**Reply by fz6m** [reply]
给最小复现看看

**Reply by lbyupup** [reply]
发现是 ```react-activation``` 这个keep-alive插件引起的，升级到最新稳定版本后解决
